src/openMVG/multiview/solver_resection_p5pfr_test.cpp

```diff
@@ -0,0 +1,630 @@
+{
+  if(expect.rows() != actual.rows() || expect.cols() != actual.cols())
+    return false;
+  for(int i = 0; i < expect.rows(); ++i)
+  {
+    for(int j = 0; j < expect.cols(); ++j)
+    {
+      if(abs(expect(i, j) - actual(i, j)) > (eps * abs(expect(i, j))))
+        return false;
+    }
+  }
+  return true;
+}
+
+bool checkNear(Vec &expect, Vec &actual, double eps)
+{
+  if(expect.rows() != actual.rows())
+    return false;
+  for(int i = 0; i < expect.rows(); ++i)
+  {
+    if(abs(expect(i) - actual(i)) >(eps * abs(expect(i))))
+      return false;
+  }
+  return true;
+}
+
+bool checkNear(Vec3 &expect, Vec3 &actual, double eps)
+{
+  if(expect.rows() != actual.rows())
+    return false;
+  for(int i = 0; i < expect.rows(); ++i)
+  {
+    if(abs(expect(i) - actual(i)) >(eps * abs(expect(i))))
+      return false;
+  }
+  return true;
+}
+
+bool checkNear(double expect, double actual, double eps)
+{
+  if(abs(expect - actual) > (eps * abs(expect)))
+    return false;
+  return true;
+}
+
+bool sortM(const resection::M &i, const resection::M &j)
+{
+  return (i._f < j._f);
+}
+
+bool testSolutions(std::vector<resection::M> &solutions, std::vector<resection::M> &models, double eps)
+{
+  if(models.size() != solutions.size())
+    return false;
+
+  for(int i = 0; i < models.size(); ++i)
+  {
+    if(!checkNear(solutions.at(i)._f, models.at(i)._f, eps))
+      return false;
+    if(!checkNear(solutions.at(i)._R, models.at(i)._R, eps))
+      return false;
+    if(!checkNear(solutions.at(i)._t, models.at(i)._t, eps))
+      return false;
+    if(!checkNear(solutions.at(i)._r, models.at(i)._r, eps))
+      return false;
+  }
+  return true;
+}
+
+TEST(Resection_P5Pfr, RandomRealExample)
+{
+  // DATA
+  Mat pt2D = Mat(2, 5);
+  Mat pt3D = Mat(3, 5);
+  pt2D << 4.803799999999999e+02, 5.422099999999998e+02, 1.508150000000000e+03, -1.640501000000000e+03, -4.026700000000001e+02, 
+          -1.326581000000000e+03, -7.138680000000001e+02, -1.234743000000000e+03, -6.222770000000000e+02, -1.461412100000000e+03;
+  pt3D << 3.175360000000000e+00, 2.531220000000000e+00, 2.420890000000000e+00, 7.578520000000000e-01, 3.880860000000000e+00, 
+          -4.679980000000000e-01, 1.964960000000000e-01, -1.863250000000000e+00, 1.648030000000000e-01, 4.177460000000000e-01, 
+          -8.605090000000000e-01, -1.507400000000000e+00, -1.828080000000000e+00, 1.208810000000000e+00, 2.980880000000000e-01;
+
+  // SOLUTIONS
+  std::vector<resection::M> solutions;
+  Mat R1 = Mat(3, 3);
+  R1 << -9.430073299811607e-01, 2.928814581476584e-01, -1.579798312288744e-01, 
+          -3.071682103315301e-01, -5.834954588038342e-01, 7.517850358427121e-01, 
+          1.280033834065588e-01, 7.574651813968776e-01, 6.402043680012413e-01;
+  Vec3 t1;
+  t1 << 3.351461753495843e+00, 3.662505952393360e-01, 1.522470547430807e+00;
+  double f1 = 8.754207183076484e+03;
+  Vec r1 = Vec(1);
+  r1 << -2.974934229711877e+01;
+  resection::M m1(R1, t1, r1, f1);
+  solutions.push_back(m1);
```

Comment:
merge this and the above line into a simpler `solutions.emplace_back(R1, t1, r1, f1)`